src/FSharp.Core/prim-types.fs

```diff
@@ -5346,6 +5349,24 @@ namespace Microsoft.FSharp.Core
             [<CompiledName("Hash")>]
             let inline hash x = GenericHash x
 
+            #if !BUILDING_WITH_LKG && !NO_NULLCHECKING_LIB_SUPPORT
+
+            [<CompiledName("NonNull")>]
+            let inline nonNull (x: 'T | null when 'T : not null and 'T : not struct) : 'T = (# "" x : 'T #)
```

Comment:
Shall we add `[<CompilerMessage>]` to it, saying that it's unchecked and neglects all the null safety? So at least first use is conscious (by ignoring it).

Reply:
How would that be different to ignoring the original warning without using `Unchecked.nonNull` ?

We do not have it for other functions in the Unchecked module as far as I can see.

Reply:
It's different because ignoring original warning will turn off nullness everywhere. This one will just acknowledge that "yes, we know when we explicitly want to strip things".